addons/survey/views/survey_templates.xml

```diff
@@ -296,7 +296,7 @@
                     </div>
                 </div>
             </div>
-            <div class="o_survey_result p-4 col-12 row" t-if="graph_data">
+            <div class="o_survey_result p-4 col-12 row" t-if="graph_data and survey.show_graph">
```

Comment:
Would be better to avoid computing graph_data then, unless it's used for something else?

Could save some of those precious queries.